google/cloud/iot/v1/device_manager.proto

```diff
@@ -31,6 +31,61 @@ option java_multiple_files = true;
 option java_outer_classname = "DeviceManagerProto";
 option java_package = "com.google.cloud.iot.v1";
 
+// TODO: I much prefer each imported_type to be declared separately,
+// rather than one with an "additional_type" repeated field.
+// TODO: Note that in the gapic_config these are actually defined as "*", not as currently in IamResource
+option (google.api.imported_type) = {
+  name: "google.iam.v1.SetIamPolicyRequest",
```

Comment:
Also, the `{name=...}` segment in the `google.api.http` annotations on those RPCs references a resource defined in the proto, yes?

Reply:
As discussed in @andreamlin's doc, I'd rather not special-case this at all by using the `name` segment. Using the `name` segment also means `http name` and resource-names are tied together in a way that may not always be true. And it adds Google-specific "knowledge" into the (some) generators that I think we want to avoid.

Also, the `name=` segment of the `http` annotation contains `*`, not `useful_name`, which will mean sensible names cannot be generated.

Reply:
I agree this could be annotated in the method rather than here, but doing it in a style like this seems more consistent with the general mechanisms in use with resource-names.
I.e. the general semantics of which fields refer to which resources is defined; then the generator can just do the right thing.